File: python/Programmers/문자열 다루기 기본.py
'''
문자열의 길이가 4 또는 6이고 모두 숫자로 이루어져있는지 판별하는 문제
함수는 return을 하면 실행이 중지되므로 조건에 맞지 않을 때 return을 해주면 
쉽게 풀이 가능
'''

def solution(s):
    answer = True
    
    if len(s) != 4 and len(s) != 6:
        answer = False
        return answer
    
    for i in range(len(s)):
        if not ('0' <= s[i] <= '9'):
            answer = False
            return answer
        
    return answer

File: Javascript/Programmers/문자열 다루기 기본.js
/*
문자열의 길이가 4 또는 6인지, 모두 숫자인지를 파악하는 문제
*/

function solution(s) {
  var answer = true;
  let len = 0
  
  for (i = 0; i < s.length; i += 1) {
      len += 1

      // isNaN 함수는 숫자가 아닐 경우 true 반환
      if (isNaN(s[i])) {
          answer = false
          break
      }
  }
  
  if (len !== 4 && len !== 6) {
      answer = false
  }
  
  return answer;
}